### ISSUE TITLE
CharacterCounter: Error message is not announced

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2240]
[A11Y-1602](https://jira.cengage.com/browse/A11Y-1602)

Platform:
Windows 11, Chrome 138, JAWS 2025
Windows 11, Edge 138, NVDA 2025

Steps to Reproduce:
1. Launch the homepage: https://faculty.cengage.com/
2. Navigate to My Library page and activate the Create Course button.
3. Now navigate the 'Create Course 2 of 2' screen using tab/arrow.
4. Now move to form section & verify.

Actual Result:
When navigating to the Course Name and Section Name fields using Tab/Shift+Tab after the error appears, the associated inline error message is not announced by the screen reader.

Expected Result:
The inline error message should be programmatically associated with the respective input field so that it is announced when the user focuses on the field.

Intsances:
1. [Accessibility] [Homepage > My Courses >Activate-Create Course-Button > Create course 4 of 4> Form Filled]: 'Course name' and 'Section name' inline error is not associated with respective edit field when user navigate using tab/shift tab.
https://app.screencast.com/qpOw7E9W8Ws3h

WCAG Guidelines:
1.3.1 Info and Relationships Level A

User Impact:
Screen reader users may not be aware of what went wrong or what needs to be corrected if the error message is not read while navigating through the form fields.

Recommendation:
Ensure the error messages are associated with their respective fields using aria-describedby and that focus behavior allows screen readers to pick up both the input label and the error text.

Screencast:
https://app.screencast.com/qpOw7E9W8Ws3h

**Note from Dashboard team:**
We’ve noticed a specific conflict when using a required input field with the maxCount prop: when maxCount is present and the input is left empty, the error message displays visually on blur/submit but is not announced by screen readers.
However, if we remove the maxCount prop, the error is announced correctly.
We think it might be on your [radar](https://github.com/cengage/react-magma/issues/1255), if so, could you please give us ETA or some suggestions for workaround?

**Comment by nikitaorliak-cengage** [comment]
**Tested on:**
- V5: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/charactercounter--default
- V4: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/charactercounter--default

Everything works as expected. No bugs found.
